client/galaxy/scripts/mvc/tool/tool-form-composite.js

```diff
@@ -319,6 +319,19 @@ define([ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view'
             }
         },
 
+        /** Refresh the history after job submission while form is shown */
+        _refreshHistory: function() {
+            var self = this;
+            var history = parent.Galaxy && parent.Galaxy.currHistoryPanel && parent.Galaxy.currHistoryPanel;
+            this._refresh_history && clearTimeout( this._refresh_history );
+            if ( history ) {
+                history.refreshContents();
+                if ( history.model.numOfUnfinishedShownContents() === 0 ) {
+                    this._refresh_history = setTimeout( function() { self._refreshHistory() }, 2000 );
+                }
+            }
+        },
+
         /** Build remaining steps */
         _execute: function() {
```

Comment:
@carlfeberhard Cool, I thought I had seen it polling.  I guess the followup question would be -- does it stop as soon as there are no other active 'new' or otherwise needing refresh jobs?

I guess you could tie it all together if, at the state of completion of one job, a new one has been created (which then forces the history to keep refreshing), but as stated we can't count on that.

As mentioned, is the WorkflowInvocation something we can represent/track in the history and base this logic on instead?